src/Text/Pandoc/Readers/LaTeX.hs

```diff
@@ -414,6 +416,12 @@ blockCommands = M.fromList $
   , "pagebreak"
   ]
 
+blockTextcolor :: PandocMonad m => LP m Blocks
+blockTextcolor = do
+    skipopts 
+    color <- braced
+    divWith ("",[],[("style","color: " ++ color)]) <$> grouped block <* notFollowedBy inline
+
```

Comment:
Yeah, this is tricky.  The problem is that we want
```
\textcolor{red}{Hi} there
```
to be parsed as a single paragraph, and
```
\textcolor{red}{
\begin{itemize}
    \item Test 1
    \item Test 2
\end{itemize}
} some more text
```
to be parsed as a colored list followed by a paragraph "some more text."  But if the block parser accepts "Hi" as a block, then it will handle these both the same.

Perhaps the thing to do is to have the block version fail if the contents of the braces can be exhaustively parsed as a sequence of inlines; that will allow the parser to fall through to the inline version.  Not very pretty, though, since we'd be parsing the same contents twice.

Slightly less ugly, maybe:  have the block version attempt to parse the bracketed stuff as inlines; if this succeeds, then parse many inlines following the textcolor command, and emit a single paragraph containing the textcolor command and the following inlines.  This way we don't double-parse.

So, something like
```
blockTextcolor = do
  skipopts
  color <- braced
  let constructor = divWith ("",[],[("style","color: " ++ color)])
  inlineContents <|> blockContents
  where inlineContents = do{ ils <- grouped inline; rest <- inlines; return (para (ils <> rest))}
        blockContents = grouped block
```